### ISSUE TITLE
Load .pfx or .p12 certificate from code?

### ISSUE THREAD

**Issue by andrekiba** [microsoft/azure-container-apps#378]
Is there a way to load (from C# code) a previously uploaded certificate in the container apps environment?
 This works great for a asp.net core app hosted on a Linux app service but seems not for ACA.
 The same app containerized is one that I want to try to run in ACA.

![c1](https://user-images.githubusercontent.com/12699760/187085568-dca9a566-bb2e-4bf2-bb43-a930592c38d6.PNG)
![c2](https://user-images.githubusercontent.com/12699760/187085572-c9f944ce-d4dd-400c-b068-ecfe3b6a9e34.PNG)

**Comment by kendallroden** [comment]
Hi @andrekiba for clarity, from your app code how are you loading the cert? Is it via a local file path? In app service do they mount the cert for you? Thanks for raising

**Comment by andrekiba** [comment]
thank you for your reply @kendallroden, yes in this way https://docs.microsoft.com/en-us/azure/app-service/configure-ssl-certificate-in-code#load-certificate-in-linuxwindows-containers

**Comment by andrekiba** [comment]
@kendallroden any update on this? Please tell me if you need more info. Thank you!

**Comment by andrekiba** [comment]
@kendallroden @anthonychu I don't think should be closed

**Comment by kendallroden** [comment]
Sorry for the delay will resurface

**Comment by anthonychu** [comment]
Certificates in Azure Container Apps are currently only for ingress TLS and are unavailable to the container app itself. If you store certs in Key Vault, you can access them in code via a managed identity.

**Comment by bowlerma** [comment]
Agree, it would be great to allow certificates held within the key vault to automatically be mounted into the container in the same way that they are for app services and for AKS using the secrets store CSI driver integration.

**Comment by HarikrishnanKM** [comment]
We recently migrated apps from Azure Web Apps to Azure Container Apps and ran into this certificate limitation. In Web Apps, uploaded certs could be accessed directly by the app, but in ACA they’re only usable for ingress TLS.  

Right now we work around this by preloading certificates into our base images, but that adds overhead and isn’t ideal. Key Vault isn’t a fit for us since we use other PAM solutions (like Delinia,Cyber Ark...).  

It would be great if ACA supported the same certificate access features as Web Apps/AKS—mounting or exposing certs securely to the container. This would make migrations smoother and reduce maintenance. Is there any timeline or plan for enabling this capability?

**Comment by bowlerma** [comment]
It's frustrating not having this capability.  We worked around it by creating a file share in the container app environment, adding the certificates to the file share and mounting this into the container app using the same location as they would be mounted when using app services.  The certificates can then be loaded from disk as normal.
